libbeat/processors/add_kubernetes_metadata/config.go

```diff
@@ -43,8 +46,52 @@ type Enabled struct {
 
 type PluginConfig []map[string]common.Config
 
+func trimAfter(value string, a string) string {
+	// Get substring after a string.
+	pos := strings.LastIndex(value, a)
+	if pos == -1 {
+		return ""
+	}
+	adjustedPos := pos + len(a)
+	if adjustedPos >= len(value) {
+		return ""
+	}
+	return value[adjustedPos:len(value)]
+}
+
+func findDaemonsConfigFlag() string {
+	// Search in processes and search for config in flags
+	psRes, _ := exec.Command("bash", "-c", "ps aux | grep -- -kubeconfig=").Output()
+	psResStr := string(psRes)
+	trimmedPsResStr := trimAfter(psResStr, "--kubeconfig=")
+	arr := strings.Split(trimmedPsResStr, " ")
+	kubeConfigPath := arr[0]
+	return kubeConfigPath
+}
+
+func getSystemKubeConfig() string {
+	homeKubeConfig := os.Getenv("HOME") + "/.kube/config"
+	if _, err := os.Stat(homeKubeConfig); !os.IsNotExist(err) {
+		return homeKubeConfig
+	}
+	envKubeConfig := os.Getenv("KUBECONFIG")
```

Comment:
I think the order use to be `env` first, then `~/.kube/config`
If you agree, can you switch these blocks?